gitpod-io/gitpod
# [server] Endpoint to issue JWT cookie, when valid session exists WEB-411

## Description


## Related Issue(s)

Fixes #

## How to test

1. Preview
2. Sign-in
3. Navigate to https://mp-server-74a8f2d94e.preview.gitpod-dev.com/api/auth/jwt-cookie, observe you get "User already has JWT session"
4. Delete JWT Cookie, try again, you'll receive a JWT cookie
5. Delete `v2` cookie, try again. you'll see "User already has a JWT session"
6. Delete both `v2` and `jwt` cookies and try again - "user has no valid session" and 401

## Documentation


#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - mp-server-74a8f2d94e</li>
	<li><b>🔗 URL</b> - <a href="https://mp-server-74a8f2d94e.preview.gitpod-dev.com/workspaces" target="_blank">mp-server-74a8f2d94e.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>
 
/hold
